src/transaction/estimate_gas.rs

```diff
@@ -6,14 +6,45 @@ use crate::future::CompatCallFuture;
 use crate::transaction::TransactionBuilder;
 use futures::compat::Future01CompatExt;
 use pin_project::pin_project;
+use serde::Serialize;
 use std::future::Future;
 use std::pin::Pin;
 use std::task::{Context, Poll};
 use web3::api::{Eth, Namespace};
 use web3::helpers::{self, CallFuture};
-use web3::types::{Address, CallRequest, U256};
+use web3::types::{Address, Bytes, U256};
 use web3::Transport;
 
+/// Transaction parameters used for estimating gas.
+///
+/// Note that this is similar to `web3::types::CallRequest` with the notable
```

Comment:
Is this an addition we could add to CallRequest upstream?

Reply:
Yes, I already created #183 tocapture that task.